src/elements/Navigation/NavigationButton.tsx

```diff
@@ -0,0 +1,81 @@
+import { cn, useComponentTheme } from '@/utils';
+import { motion } from 'motion/react';
+import React, { FC, PropsWithChildren } from 'react';
+import { NavigationTheme } from './NavigationTheme';
+
+interface NavigationButtonProps extends PropsWithChildren {
+  /**
+   * Custom class names for the navigation button.
+   */
+  className?: string;
+
+  /**
+   * Variant of the navigation button.
+   */
+  variant?: 'button' | 'underline';
```

Comment:
can this be `filled` instead of `button`? It feels more inline with what we already have for `button` variants. After all, the component is a button either way, right?

Reply:
Actually it looks like more `ghost` than `filled`, because only icon and text visible in regular state.

Reply:
makes sense. i'm good with `ghost`